### ISSUE TITLE
Better handling when runCommand tries to save the executing machine

### ISSUE THREAD

**Issue by Alaisx** [MightyPirates/OpenComputers#1384]
When issuing a WorldEdit command via the debug card, the command will fail if the affected region encloses the computer. This makes sense logically, but for some commands, such as //fill and //fillr, it is not possible to prevent the region from overlapping the computer. These commands fill a region starting at the block containing the player's feet and propagate outward and downward but never upward. A couple of things might fix this, both involving moving the fakeplayer. 

1) Add a command to move the fake player. (Not sure if this is possible, would be exceptionally helpful as most WE commands expect/require the player to move. Currently this is done by moving the computer with a second computer.)
2) Spawn the fake player with their head inside the computer, rather than the feet, so that the feet are below the computer and there is no overlap

The error message:
"§cPlease report this error: [See console]\njava.lang.reflect.InvocationTargetException: null"
From the console:
http://pastebin.com/A9MvczxF

**Comment by gjgfuj** [comment]
Doesnt worldedit let you set positions manually? I swear it did.

**Comment by Alaisx** [comment]
You can set positions for your selection manually, but many common commands do not act on the current selection. e.g. while //cut and //copy are selection based, //paste is not (usage is //copy, walk offset distance, //paste). Anything that requires a single reference point uses the player's feet position. This is because worldedit needs a single point not a multiblock selection. e.g. shape commands such as //cyl (generates cylinder with base centre at your feet) and //fill (fills a hole by propagating blocks downward and outward from your feet). 

Basically, without moving the player/computer, you are limited to commands under the "Regions" section here: http://wiki.sk89q.com/wiki/WorldEdit

It is possible to work around this by physically moving the computer (e.g. for //paste), but anything that always creates a block at the player's feet (like //fill) is currently not possible via OC.

**Comment by fnuecke** [comment]
The actual issue here is that you try to modify the world in a way that the computer tries to save itself while it is executing, which is impossible. So the only way to work around that would be to allow queuing commands to be executed independently at some later time; something like `runCommandAsync` maybe?

**Comment by Alaisx** [comment]
I see, so it's impossible to have the fake player/command initiation located somewhere other than the computer's block? I'm probably misunderstanding something fundamental about how the commands are executed...

Delaying the execution would allow moving the computer out of the way before it gets 'caught' in the region to be modified. It's just a little more convoluted, since it adds a delay and requires a second computer. Any solution is good though.

**Comment by fnuecke** [comment]
Ah, sorry, had missed the your second post (only saw the notification mail for the first one). Hrm, having a fake position for the fake player could work then, I suppose.

**Comment by natedogith1** [comment]
Couldn't you use worldedit's /toggledplace command? It switches whether the "origin" is the player or the selected pos1.

**Comment by Alaisx** [comment]
I didn't even know you could do that. Thanks so much for solving my problem, and I am sorry for not figuring it out myself. I will let you know if I find anything that this method fails on. 

You are welcome to close this issue if you like, unless you want to add something like error handling so the computer doesn't give you "please report this issue". I imagine there are lots of ways RunCommand could break when calling external mods' commands, so maybe it's not that easy to fix.

Apologies again for apparently not doing my homework!

**Comment by fnuecke** [comment]
Since this has been solved, and the debug card is, well, sortakinda unsupported anyway, trying to work around this for a nice message is more effort than I can spare so eh :P
